### PR TITLE
Add endpoint refactor

### PR DESCRIPTION
**Pull request by McGarvs**
- Made some schema changes to check ObjectIds, makes some code redundant, but does not break anything
- Add New Engagement from scratch should work
- Add New TestPlan from scratch should work
- Set Active TestPlan should work (fixed a lot of edge cases)
- Add New TestCase from scratch should work
- Add New Test from scratch should work
- Made some similar code refactors to each of the edit buttons, those should still work

Did not finish refactoring all of the add API's, this is big enough already. Will do that in a part 2 PR.

### CODE REVIEW

File: schemas/engagementSchema.js [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+  return (isOptional ? {
+    name: name,
+    message: (name + " is not a valid ObjectId"),
+    test: val => val === undefined || ObjectId.isValid(val)
+  } : {
+    name: name,
+    message: (name + " is not a valid ObjectId"),
+    test: val => ObjectId.isValid(val)
+  })
+}
```
Comment (Jkwok2018) [inline schemas/engagementSchema.js:14]: Since this function is in every schema, why can't we export it somewhere and then each schema can just import it?

**Review by Jkwok2018 — COMMENTED** [review]
Except for the redundant objectIdTest function, everything else looks good to me
